chromium_src/chrome/browser/ui/bookmarks/bookmark_utils.cc

```diff
@@ -18,8 +24,24 @@
 #define GetBookmarkFolderIcon GetBookmarkFolderIcon_UnUsed
 #endif
 
+#define ToggleBookmarkBarWhenVisible                                  \
+  ToggleBookmarkBarWhenVisible_ChromiumImpl(                          \
+      content::BrowserContext* browser_context);                      \
+  void ToggleBookmarkBarWhenVisible(                                  \
+      content::BrowserContext* browser_context) {                     \
+    PrefService* prefs = user_prefs::UserPrefs::Get(browser_context); \
+    const bool always_show =                                          \
```

Comment:
I'm confused, in the original implementation always_show is `!prefs->GetBoolean(bookmarks::prefs::kShowBookmarkBar)`

Reply:
solved by moving interaces to serpate file